clusterloader2/docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,303 @@
+So our template for deployment will look like this (`deployment.yaml` file):
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Name}}
+  labels:
+    group: test-pod
+spec:
+  replicas: {{.Replicas}}
+  selector:
+    matchLabels:
+      group: test-pod
+  template:
+    metadata:
+      labels:
+        group: test-pod
+    spec:
+      containers:
+      - image: k8s.gcr.io/pause:3.1
+        imagePullPolicy: IfNotPresent
+        name: {{.Name}}
+        resources:
+          requests:
+            cpu: 1m
+            memory: 10M
+```
+## Execute test
+Before running test, make sure that kubeconfig current context points to kind cluster:
+```bash
+kubectl config current-context
+kind-test-cluster
+```
+To execute test, run:
+```bash
+cd clusterloader2/
+go run cmd/clusterloader.go --testconfig=config.yaml --provider=kind --kubeconfig=${HOME}/.kube/config --v=2
+```
+
+At the end of clusterloader output you should see pod startup latency:
+```json
+{
+  "data": {
+    "Perc50": 7100.534796,
+    "Perc90": 8702.523037,
+    "Perc99": 9122.894555
+  },
+  "unit": "ms",
+  "labels": {
+    "Metric": "pod_startup"
+  }
+},
+```
+`pod_startup` measures time since pod was created until it was observed via watch as running.
+
+You should also see that test succeeded:
+```
+--------------------------------------------------------------------------------
+Test Finished
+Test: ./config.yaml
+Status: Success
+--------------------------------------------------------------------------------
+```
+As an exercise you can modify threshold for PodStartupLatency below values you've observed in your run and check if test fails.
+## Delete kind cluster
+To delete kind cluster, run:
+```bash
+kind delete cluster --name test-cluster
+```
+
+## Running 100-node scale test
+
+Here you can find general purpose [Load test]. This test is release-blocking test we use to evaluate scalability of kubernetes.
+It consists of 3 main phases:
+- Creating objects
+- Scaling objects to size between (50%, 150%) of their original size
+- Deleting objects
+
+It can be used to test clusters starting from 100 nodes up to 5k nodes.
+Load test will create, roughly 30 * nodes pod objects. It will create:
+- deployments
+- jobs
+- statefulsets
+- services
+- secrets
+- configmaps
+
+There are small (5 pods), medium (30 pods) and big (250 pods) versions of them.
+
+First, you need to create 100-nodes cluster then you can run cluster scale test with this command:
+```bash
+./run-e2e-with-prometheus-fw-rule.sh cluster-loader2 --testconfig=./testing/load/config.yaml --nodes=100 --provider=gke --enable-prometheus-server=true --kubeconfig=${HOME}/.kube/config --v=2
+```
+
+`--enable-prometheus-server=true` creates prometheus server using prometheus-operator.
+
+There are various measurements that depend on prometheus metrics, for example:
+- API responsiveness - measures the latency of requests to kube-apiserver
+- Scheduling throughput
+- NodeLocalDNS latency
+
```

Comment:
I'm missing some summary at the end (e.g. next steps, further reading, what should be added, etc.).

Reply:
I will leave it for now. I will add it once we have second tutorial with developing your measurement. Also, I am thinking about adding some "contributing" guide.